### PR TITLE
core: add `AllowHostDirVolumePlugin: true` to SCC and remove unused imports

### PR DESCRIPTION
**Pull request by sp98**
Signed-off-by: Santosh Pillai <sapillai@redhat.com>

**Description of your changes:**
- Adds `AllowHostDirVolumePlugin` to the `go` representation of the rook-ceph SCC for using the hostPath. Also making it consistent with the yaml representation. 
- Removes unused import.

From Openshift [Docs](https://docs.openshift.com/container-platform/4.9/authentication/managing-security-context-constraints.html):
``` 
For backwards compatibility, the usage of allowHostDirVolumePlugin overrides settings in the volumes field. For example, if allowHostDirVolumePlugin is set to false but allowed in the volumes field, then the hostPath value will be removed from volumes.
```

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: pkg/apis/ceph.rook.io/v1/scc.go [hunk 1]
```diff
@@ -17,7 +17,6 @@ limitations under the License.
 package v1
 
 import (
-	_ "embed"
```
Comment (leseb) [inline pkg/apis/ceph.rook.io/v1/scc.go:20]: This should be a separate commit.

**Review by leseb — CHANGES_REQUESTED** [review]
The commit message is empty...

File: pkg/apis/ceph.rook.io/v1/scc.go (hunk 1 above)
Comment (sp98) [reply pkg/apis/ceph.rook.io/v1/scc.go:20]: moved to a separate commit.

**Review by leseb — COMMENTED** [review]
Why is the commit message different from the PR description?

**Comment by sp98** [comment]
> Why is the commit message different from the PR description?

updated the PR description.

File: pkg/apis/ceph.rook.io/v1/scc.go [hunk 2]
```diff
@@ -38,6 +37,7 @@ func NewSecurityContextConstraints(name, namespace string) *secv1.SecurityContex
 			Namespace: namespace,
 		},
 		AllowPrivilegedContainer: true,
+		AllowHostDirVolumePlugin: true,
```
Comment (leseb) [inline pkg/apis/ceph.rook.io/v1/scc.go:40]: Why do we need to enable this?

**Review by leseb — CHANGES_REQUESTED** [review]
The commit message hasn't been updated, the `From Openshift Docs:` section is still missing.

File: pkg/apis/ceph.rook.io/v1/scc.go (hunk 2 above)
Comment (sp98) [reply pkg/apis/ceph.rook.io/v1/scc.go:40]: mon canary pods won't come up if we don't enable this enabled.  Rook logs show : 
```
ceph-cluster-controller: failed to get ceph daemons versions, this typically happens during the first cluster initialization. failed to run 'ceph versions'. . unable to get monitor info from DNS SRV with service name: ceph-mon2021-11-08T11:19:26.938+0000 7f9fb7fff700 -1 failed for service _ceph-mon._tcp
```

**Comment by sp98** [comment]
> The commit message hasn't been updated, the `From Openshift Docs:` section is still missing.

meant to add it to the PR description only.

**Comment by leseb** [comment]
> > The commit message hasn't been updated, the `From Openshift Docs:` section is still missing.
> 
> meant to add it to the PR description only. Do you suggested having that note in the commit as well?

Why? If it's useful in the PR description, it's useful as a commit message too. The PR description must be **identical** to the commit message.

File: pkg/apis/ceph.rook.io/v1/scc.go (hunk 2 above)
Comment (leseb) [reply pkg/apis/ceph.rook.io/v1/scc.go:40]: I don't understand how the error relates to this setting. Isn't this an env issue?

Comment (travisn) [reply pkg/apis/ceph.rook.io/v1/scc.go:40]: > I don't understand how the error relates to this setting. Isn't this an env issue?

All of the ceph daemons have a hostPath volume for the logs, right? So the hostPath volume is needed or else all the daemon pods would fail to start. This setting also matches what we already have in the [operator-openshift.yaml](https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/operator-openshift.yaml#L14).

Comment (leseb) [reply pkg/apis/ceph.rook.io/v1/scc.go:40]: Ah right, I always think of flex when I see this I don't know why. Makes sense.

**Review by leseb — APPROVED** [review]
